### PR TITLE
Mutation refetch queries with variables and update

### PR DESCRIPTION
**Pull request by MargaretKrutikova**
This PR addresses #26. Summary:
- adds possibility to pass query variables in `refetchQueries` in mutation options,
- adds `update` to mutation options,
- adds some utility functions to convert to `queryObj` and `readQueryOptions` from `reason-apollo`,
- adds example of using `refetchQueries` and `update` for manually updating `apollo` cache.
---
This PR changes the type of `refetchQueries` from `array(string)` to 
```reason
ReasonApolloTypes.executionResult => array(ApolloClient.queryObj)
```
 which is a breaking change. So all cases with `~refetchQueries=[|"getAllPersons"|]` need to be replaced with:

```reason
let getAllPersons = Persons.GetAllPersons.make();
...
~refetchQueries= _ => [|{ 
   "query": ApolloClient.gql(. getAllPersons##query),
   "variables": getAllPersons##variables,
}|]
```
or with a utility function:

```reason
~refetchQueries= _ => {
    let getAllPersons = Persons.GetAllPersons.make();
    [|ReasonApolloHooks.Utils.toQueryObj(query)|];
},
```
Would that be acceptable?

While I was at it, I also added the `update` to mutation options 🙂 It is defined [here](https://github.com/apollographql/apollo-client/blob/master/packages/apollo-client/src/core/watchQueryOptions.ts#L279) in `apollo-client` and can be used, for example, to update cache manually after a mutation has returned.

I also struggled with duplicated dependencies on `react` in the example project, which resulted in runtime errors because `yarn` and `npm` seem to duplicate `peerDependencies` when developing locally. I added a half hackish solution with using `npm dedupe` on `reason-react` which seem to work. Does anyone have a good solution to this problem?

### CODE REVIEW

**Comment by fakenickels** [comment]
This is really cool, I liked the single function usage better also!
I'm gonna take a better look at the whole code soon. Thanks for your contributions @MargaretKrutikova !

**Comment by MargaretKrutikova** [comment]
Thank you! It is a lot of fun contributing to this project, so I am happy I have the opportunity 🙂

**Comment by Yakimych** [comment]
Great timing - just tested it when I was implementing a mutation with two refetch queries with parameters. Worked like a charm!

File: examples/persons/src/FilterByNameCache.re [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    allPersons(filter: { name: $name } ) {
+      id
+      name
+      age
+    }
+  }
+|}
+];
+
+external cast: Js.Json.t => PersonsNameFilterConfig.t = "%identity";
```
Comment (fakenickels) [inline examples/persons/src/FilterByNameCache.re]: thanks for this advanced demo!
Could you add some more comments about what is going on and why the casting and raw are necessary for this very cases? Probably some beginner will read this in the future as reference

**Comment by fakenickels** [comment]
I'm going to merge and then we can add those comments

**Comment by fakenickels** [comment]
Thanks a lot again for your contribution @MargaretKrutikova !

**Comment by fakenickels** [comment]
Released under v2.5.0

File: examples/persons/src/FilterByNameCache.re (hunk 1 above)
Comment (MargaretKrutikova) [reply examples/persons/src/FilterByNameCache.re]: Yes, absolutely! I will prepare a PR with some comments on the source code.

**Pull request merged**